portfolio/src/main/webapp/src/components/BookService.js

```diff
@@ -1,54 +1,128 @@
 import React, { Component } from "react";
 import Button from "@material-ui/core/Button";
 import { Link } from "react-router-dom";
-import Calendar from 'react-calendar';
+import Calendar from "react-calendar";
 import SearchBar from "./SearchBar";
 
 export default class BookService extends React.Component {
-    constructor() {
-        super();
-        this.state = {
-            date: new Date(),
-            booking_name: "",
-            booking_date: "",
-            booking_duration: "",
-            booking_description: "",
-            booking_price: "",
-        };
-    }
+  constructor() {
+    super();
+    this.state = {
+      booking_name: "Teach Modern Computer Skills to Elderly",
+      //booking_date: new Date(),
+      booking_date: "",
+      booking_duration: "",
+      booking_optional_note: "",
+      booking_price: "",
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.onChangeCalendar = this.onChangeCalendar.bind(this);
+  }
 
-    onChangeCalendar = date => this.setState({ date })
+  onChangeCalendar = (booking_date) => this.setState({ booking_date });
 
-    render() {
-        return (
-            <div class="book-service-all-container">
-                <div class="book-service-header-container">
-                    <div class="logo-container">theCOMMONS PROJECT</div>
-                    <div class="searchbar-container"><SearchBar /></div>
-                    <div class="options-container">
-                        <Link to="/dashboard">Dashboard</Link>
-                        <Link to="/profile"></Link>
-                        <button>Logout</button>
-                    </div>
+  handleChange(event) {
+    const { name, value, type, checked } = event.target;
+    this.setState({
+      [name]: value,
+    });
+  }
+
+  render() {
+    return (
+      <main>
+        <form action="/book-new-service" method="POST">
```

Comment:
Are you sure you want to change this into a form, instead of saving changes in state and then submitting axios request?

Reply:
I think a form makes sense because it makes it compatible with the servlet that's already been made!